torch/lib/ATen/dlpack.h

```diff
@@ -114,6 +114,19 @@ typedef struct {
   /*! \brief The offset in bytes to the beginning pointer to data */
   uint64_t byte_offset;
 } DLTensor;
+
+/*!
+ * \ manage memory of DLTensor
+ */
+typedef struct {
+  /*! DLTensor which is being memory managed*/
+  DLTensor dlTensor;
+  /* context in which DLTensor is used in framework*/
+  void * ctx;
+ /* Destructor signature */
+  void (*destructor)(void * ctx, void * deleter_arg);
+} DLManagedTensor;
+
```

Comment:
Also, is it a good idea to add things to this file? It seems that it's copy pasted from dlpack. What if, sometime in the future, it's going to change? We'll have to manually patch the parts we modified.

Reply:
Yes, I'd suggest moving the new struct to an extra little header. Doesn't cost much.

Reply:
This needs to eventually make it back to dlpack. The point is that frameworks have to _agree_ on destructor approach used in dlpack, otherwise why use DLPack? We could instead just make direct converters from PyTorch <-> C2 and everything would be simpler.